### PR TITLE
chore: add release workflow on pr

### PR DESCRIPTION
**Pull request by tejal29**
Should have set it up before.

### CODE REVIEW

File: .github/workflows/pr_release.yaml [hunk 1]
```diff
@@ -0,0 +1,139 @@
...
+        version: latest
+       
+
+    - uses: docker/build-push-action@v2
+      with:
+        context: .
+        file: ./deploy/Dockerfile_slim
+        platforms: linux/amd64,linux/arm64,linux/s390x,linux/ppc64le
+        tags: |
+          gcr.io/kaniko-project/executor:${{ env.GITHUB_SHA }}-slim
```
Comment (mattmoor) [inline .github/workflows/pr_release.yaml:50]: We definitely shouldn't be pushing to this project on PRs 🤔 

cc @dlorenc

File: .github/workflows/pr_release.yaml [hunk 2]
```diff
@@ -0,0 +1,139 @@
+name: Build images on pull requests
+
+on: [pull_request]
```
Comment (mattmoor) [inline .github/workflows/pr_release.yaml:3]: secrets are not available to `pull_request` but they are available to `pull_request_target` (the script is executed from the base of the PR though)

File: .github/workflows/pr_release.yaml [hunk 3]
```diff
@@ -0,0 +1,139 @@
...
+
+    - name: Set up Docker Buildx
+      id: buildx
+      uses: docker/setup-buildx-action@v1
+      with:
+        version: latest
+       
+
+    - uses: docker/build-push-action@v2
+      with:
```
Comment (mattmoor) [inline .github/workflows/pr_release.yaml:45]: You need to run the docker login action prior to pushing

Comment (mattmoor) [reply .github/workflows/pr_release.yaml:45]: I guess the other file does this with:
```
    - name: Setup gcloud CLI
      uses: google-github-actions/setup-gcloud@master
      with:
        service_account_key: ${{ secrets.GCR_DEVOPS_SERVICE_ACCOUNT_KEY }}
        project_id: kaniko-project
        export_default_credentials: true
    # Configure docker to use the gcloud command-line tool as a credential helper
    - run: |
        # Set up docker to authenticate
        # via gcloud command-line tool.
        gcloud auth configure-docker
```

Comment (mattmoor) [reply .github/workflows/pr_release.yaml:45]: However, as I mention above, you cannot access `secrets.GCR...` in `pull_request` workflows, nor do I think we should be pushing things from PRs (especially to the prod project).

File: .github/workflows/pr_release.yaml (hunk 1 above)
Comment (tejal29) [reply .github/workflows/pr_release.yaml:50]: yeah push as well as signing is turned off now for project PRs.

File: .github/workflows/pr_release.yaml (hunk 2 above)
Comment (tejal29) [reply .github/workflows/pr_release.yaml:3]: yep. I read the fine print 4 hours late :(

**Pull request merged**